src/main/java/bio/terra/service/filedata/azure/AzureSynapsePdao.java

```diff
@@ -552,6 +551,67 @@ public void createSnapshotRowIdsParquetFile(
     executeSynapseQuery(sqlCreateRowIdTable.render());
   }
 
+  public Map<String, Long> createSnapshotParquetFilesByQuery(
+      AssetSpecification assetSpec,
+      UUID snapshotId,
+      String datasetDataSourceName,
+      String snapshotDataSourceName,
+      String translatedQuery,
+      Boolean isGlobalFieldIds)
+      throws SQLException {
+    Map<String, Long> tableRowCounts = new HashMap<>();
+
+    // First handle root table
+    AssetTable rootTable = assetSpec.getRootTable();
+    String rootTableName = rootTable.getTable().getName();
+
+    ST sqlCreateSnapshotTableTemplate = new ST(createSnapshotTableByQueryTemplate);
+    ST queryTemplate =
+        generateSnapshotParquetCreateQuery(
+            sqlCreateSnapshotTableTemplate,
+            IngestUtils.getSourceDatasetParquetFilePath(rootTable.getTable().getName()),
+            rootTable.getTable().getName(),
+            snapshotId,
+            IngestUtils.getSnapshotSliceParquetFilePath(snapshotId, rootTableName, "root"),
+            datasetDataSourceName,
+            snapshotDataSourceName,
+            rootTable.getSynapseColumns(),
+            isGlobalFieldIds);
```

Comment:
This appears to duplicate the code in `createSnapshotParquetFilesByAsset()`, if so, is there a way to share the code instead?

Reply:
I don't think there is a good way to do this.
L 562-566 creates items that are used elsewhere in this method, so we need them defined even if we pulled them into a shared method.
L568 is unique to the snapshot create type
So, that leaves us with L569, which is already a call to shared code.